tasks/rolling_restart.yml

```diff
@@ -55,10 +55,10 @@
 - name: Wait for all pods to be ready again
   ansible.builtin.shell: |
     set -o pipefail
-    {{ rke2_data_path }}/bin/kubectl --kubeconfig /etc/rancher/rke2/rke2.yaml get pods -A --field-selector=metadata.namespace!=kube-system | grep -iE "crash|error|init|terminating" | wc -l
+    {{ rke2_data_path }}/bin/kubectl --kubeconfig /etc/rancher/rke2/rke2.yaml get pods -A --field-selector=metadata.namespace!=kube-system,status.phase!=Running,status.phase!=Succeeded --ignore-not-found | wc -l
```

Comment:
As per your suggestion, please remove the kube-system exception from the check command.

Reply:
Removed